gin_integration_test.go

```diff
@@ -561,3 +561,25 @@ func TestTreeRunDynamicRouting(t *testing.T) {
 func isWindows() bool {
 	return runtime.GOOS == "windows"
 }
+
+func TestEscapedColon(t *testing.T) {
+	router := New()
+	f := func(u string) {
+		router.GET(u, func(c *Context) { c.String(http.StatusOK, u) })
+	}
+	f("/r/r\\:r")
```

Comment:
Should probably add a test case for `\:r` as well, which should not result in expected behaviour.

```
f("/r/r/\:r")
```

Reply:
I tried, but got a compilation error `unknown escape sequence`

seems the `f("/r/r/\:r")` contains an invalid string literal